storytower/src/components/StoryDetails/StoryDetails.jsx

```diff
@@ -43,14 +61,18 @@ const StoryDetails = () => {
                     <p>Chapter Count: {story.chapterCount}</p>
                     <p>Synopsis: {story.synopsis}</p>
                     <p>Genres: {story.genres.join(', ')}</p>
-
+                    {isLoggedIn && (
+                        <button onClick={handleBookmarkClick}>
+                            {isBookmarked ? 'Unbookmark' : 'Bookmark'}
```

Comment:
some way to 
a) make sure that the state displayed always matches the state accurately
b) prevent a user from firing a hundred db calls by spamming this button

Reply:
Fixed the first one, not the second one. Since this PR is getting out of scope, I'm going to merge it and make another branch specifically for bookmark features.